### DISCUSSION TITLE
It would be very helpful to have the Istio resource definitions published as JSONSchema

### DISCUSSION THREAD

**Discussion by InfoSec812** [istio/istio#44635] [Ideas]
I often need to write Operators, Helm Charts, etc... which can be enhanced with JSONSchema validation. Right now, I cannot find a JSONSchema document published by the Istio community but I have found some published by [external parties](https://github.com/snowdrop/istio-java-api/blob/master/istio-model/src/main/resources/schema/). I worry that the externally maintained schema could end up out of sync and would love it if we could publish JSONSchema for all of the Istio resource types and children. Thoughts?

**Comment by craigbox** [comment]
It looks like this is being addressed (or not) in https://github.com/istio/istio/issues/33535.

**Reply by InfoSec812** [reply]
Thanks for the link. This just seems like something that should be simple. I see that upstream is considering supporting `$ref` in the Schema for CRDs, and this would allow for CRDs to broken up into smaller components. https://github.com/kubernetes/kubernetes/issues/82292

**Reply by howardjohn** [reply]
To be clear, we arelady have this for everything _but_ IstioOperator. All the core types like VirtualService, etc have fully specced out Openapi specs defined (in the CRD).
